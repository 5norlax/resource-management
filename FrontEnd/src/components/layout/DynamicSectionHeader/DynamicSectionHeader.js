import React from 'react';
import { Button, Row } from 'reactstrap';

export default React.memo(function DynamicSectionHeader({ name, count, addForm}) {
    const AddFormButton = (count) => {
        while (count < 3) return <Button color="primary" className="section-header shadow-none" onClick={addForm}>
            <i style={{ color: 'inherit' }} className="fas fa-plus-circle fa-lg"></i>&nbsp;&nbsp;Add New {name}</Button>
    }
    return (
        <div>
            <Row>
            <div className="col-md-9">
                <h3 data-testid="dynamic-header-name">{name}</h3>
            </div>
            <div className="col-md-3">
                {AddFormButton(count)}
            </div>
            </Row>
            <hr />
            
        </div>
    )
});