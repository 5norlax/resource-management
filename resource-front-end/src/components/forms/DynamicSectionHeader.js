import React from 'react';
import { Button } from 'reactstrap';

const DynamicSectionHeader = ({ name }) => {
    return (
        <div>
            <div style={{ display: 'flex', justifyContent: 'space-between' }}>
                <h3>{name}</h3>
                <Button style={{}}><i className="fas fa-plus-circle"></i>+ Add New {name}</Button>
            </div>
            <hr />
        </div>
    )
}

export default DynamicSectionHeader