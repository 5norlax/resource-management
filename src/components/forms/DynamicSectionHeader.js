import React from 'react';
import { Button } from 'reactstrap';

const DynamicSectionHeader = ({ name, count, addForm}) => {
    const AddFormButton = (count) => {
        while (count < 3) return <Button onClick={addForm} style={{}}><b>+</b> Add New {name}</Button>
    }
    return (
        <div>
            <div style={{ display: 'flex', justifyContent: 'space-between' }}>
                <h3>{name}</h3>
                {AddFormButton(count)}
            </div>
            <hr />
        </div>
    )
}

export default DynamicSectionHeader