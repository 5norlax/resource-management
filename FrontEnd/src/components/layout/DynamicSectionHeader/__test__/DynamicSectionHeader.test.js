import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import DynamicSectionHeader from '../DynamicSectionHeader';

afterEach(cleanup);

describe('Test DynamicSectionHeader', () => {

  it('matches snapshot', () => {
    const tree = renderer.create(<DynamicSectionHeader />).toJSON()
    expect(tree).toMatchSnapshot();
  });

});