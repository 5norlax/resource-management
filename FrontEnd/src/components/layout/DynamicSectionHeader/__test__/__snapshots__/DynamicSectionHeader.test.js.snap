// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DynamicSectionHeader matches snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-md-9"
    >
      <h3
        data-testid="dynamic-header-name"
      />
    </div>
    <div
      className="col-md-3"
    />
  </div>
  <hr />
</div>
`;
